integrations/atlassian/confluence/client.go

```diff
@@ -90,17 +92,72 @@ func connTest(i *integration) sdkintegrations.OptFn {
 			return sdktypes.InvalidStatus, err
 		}
 
-		baseURL, err := apiBaseURL()
-		if err != nil {
-			return sdktypes.NewStatus(sdktypes.StatusCodeError, err.Error()), nil
+		at := vs.Get(authType)
+		if !at.IsValid() || at.Value() == "" {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "Init required"), nil
 		}
 
-		token := vs.GetValueByString("oauth_AccessToken")
+		switch at.Value() {
+		case integrations.OAuth:
+			err := oauthConnTest(vs)
+			if err != nil {
+				return sdktypes.NewStatus(sdktypes.StatusCodeError, err.Error()), nil
+			}
+		case integrations.APIToken:
+			err := apiTokenConnTest(nil, vs)
+			if err != nil {
+				return sdktypes.NewStatus(sdktypes.StatusCodeError, err.Error()), nil
+			}
```

Comment:
See how `connStatus` handles unrecognized values:
```suggestion
			}
		default:
			return sdktypes.NewStatus(sdktypes.StatusCodeError, "Bad auth type"), nil
```